src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,13 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+    private final Vector<String> INT_DECODE_VECTOR = new Vector<String>(10000);
```

Comment:
Is the allocation of the String[]/Vector<String> really a bottleneck here, or would it be better to avoid having this field in the class?

A Vector<String> could be allocated within the decodeInts method, in which case the method could be made static.

Furthermore, why not use an ArrayList instead of a Vector in that case, and avoid the synchronization overhead?